app/dash_app.py

```diff
@@ -1,383 +1,397 @@
+    'Edziza_North_3M13': [57.74, -130.64],
+    'Edziza_North_3M41': [57.74, -130.64],
+    'Edziza_South_3M12': [57.64, -130.64],
+    'Edziza_South_3M31': [57.64, -130.64],
+    'Edziza_South_3M42': [57.64, -130.64],
+    'Tseax_3M7': [55.11, -128.90],
+    'Tseax_3M9': [55.11, -128.90],
+    'Tseax_ 3M19': [55.11, -128.90],
+    'Tseax_3M40': [55.11, -128.90],
+    'Nazko_3M9': [52.93, -123.73],
+    'Nazko_3M20': [52.93, -123.73],
+    'Nazko_3M31': [52.93, -123.73],
+    'Hoodoo_3M8': [56.77, -131.29],
+    'Hoodoo_3M11': [56.77, -131.29],
+    'Hoodoo_3M14': [56.77, -131.29],
+    'Hoodoo_3M38': [56.77, -131.29],
+    'Hoodoo_3M41': [56.77, -131.29],
+    'LavaFork_3M11': [56.42, -130.85],
+    'LavaFork_3M20': [56.42, -130.85],
+    'LavaFork_3M21': [56.42, -130.85],
+    'LavaFork_3M29': [56.42, -130.85],
+    'LavaFork_3M31': [56.42, -130.85],
+    'LavaFork_3M41': [56.42, -130.85],
+}
+
+# could this be handled instead with a style sheet?
+LR_MARGIN_PX = 10
+TB_MARGIN_PX = 5
+
+
+def _read_coherence(coherence_csv):
+    coh = pd.read_csv(
+        coherence_csv,
+        parse_dates=['Reference Date', 'Pair Date'])
+    coh.columns = ['first_date', 'second_date', 'coherence']
+    wrong_order = (coh.second_date < coh.first_date) & coh.coherence.notnull()
+    if wrong_order.any():
+        raise RuntimeError(
+            'Some intereferogram dates not ordered as expected:\n' +
+            coh[wrong_order].to_string())
+    return coh
+
+
+def _valid_dates(coh):
+    return coh.first_date.dropna().unique()
+
+
+def _coherence_csv(target_id):
+    site, beam = target_id.split('_', 1)
+    return f'Data/{site}/{beam}/CoherenceMatrix.csv'
+
+
+def _plot_coherence(coh_long):
+    coh_long['delta_days'] = (coh_long.second_date -
+                              coh_long.first_date).dt.days
+    coh_wide = coh_long.pivot(
+        index='delta_days',
+        columns='second_date',
+        values='coherence')
+
+    # because hovertemplate 'f' format doesn't handle NaN properly
+    coh_wide = coh_wide.round(2)
+
+    coh_wide = coh_wide.loc[~(coh_wide.shift().isnull() &
+                              coh_wide.isnull()).all(axis=1)]
+    groups = (coh_wide.index.to_series().diff() > GROUP_GAP_DAYS).cumsum()
+    coh_wide.loc[0, :] = np.NaN
+    coh_wide.sort_index(inplace=True)
+
+    heights = [count/groups.value_counts().sum()
+               for count in reversed(groups.value_counts())]
+    fig = make_subplots(
+        rows=groups.nunique(), cols=1, shared_xaxes=True,
+        vertical_spacing=0.05, row_heights=heights,
+        y_title='Delta [days]')
+
+    for group, subset in coh_wide.groupby(groups):
+        row = int(groups.nunique() - group)
+        fig.add_trace(
+            Heatmap(
+                z=subset.values,
+                x=subset.columns,
+                y=subset.index,
+                hovertemplate=(
+                    'Second: %{x}<br>'
+                    'First: -%{y} days<br>'
```

Comment:
Would like to see both dates on the hover in addition to time between dates. This is useful but there are many scenarios where the analyst will want to know if the interferogram displayed covers a specific event or date. Datetime module should be able to calculate this easily with what's already here. Let's keep the temporal baseline as an additional piece of info. The integer doesn't need to be negative.

Reply:
Agreed that showing the date would be better - it's going to take a bit of work to implement that, though. I don't think it's going to be possible to calculate it on the fly, I think you need to supply the extra data using the `customdata` attribute as described at https://plotly.com/python/hover-text-and-formatting/.

The negative sign is to indicate that the first date is always before (less than) the second. I thought it was clearer than just giving the delta.

Reply:
Added issue #14 to be resolved by a future pull request.